chunk_arrow.go

```diff
@@ -18,37 +20,44 @@ type arrowResultChunk struct {
 	allocator        memory.Allocator
 }
 
-func (arc *arrowResultChunk) decodeArrowChunk(rowType []execResponseRowType, highPrec bool) ([]chunkRowType, error) {
+func (arc *arrowResultChunk) decodeArrowChunk(ctx context.Context, rowType []execResponseRowType, highPrec bool) ([]chunkRowType, error) {
 	logger.Debug("Arrow Decoder")
 	var chunkRows []chunkRowType
-
-	for {
-		record, err := arc.reader.Read()
-		if err == io.EOF {
-			return chunkRows, nil
-		} else if err != nil {
+	if arrowRecordChan := getArrowRecordChan(ctx); arrowRecordChan != nil {
+		numRows, err := arc.writeToArrowChan(arrowRecordChan)
+		if err != nil {
 			return nil, err
 		}
-
-		numRows := int(record.NumRows())
-		columns := record.Columns()
-		tmpRows := make([]chunkRowType, numRows)
-
-		for colIdx, col := range columns {
-			destcol := make([]snowflakeValue, numRows)
-			if err = arrowToValue(&destcol, rowType[colIdx], col, highPrec); err != nil {
+		chunkRows = append(chunkRows, make([]chunkRowType, numRows)...)
+		return chunkRows, nil
+	} else {
```

Comment:
this else is not needed, you have an early return. This will avoid the extra indentation and keep the original code untouched.